src/main/java/net/countercraft/movecraft/combat/features/directors/AADirectors.java

```diff
@@ -97,14 +98,15 @@ private void processFireball(@NotNull SmallFireball fireball) {
         double speed = fireballVector.length() ; // store the speed to add it back in later, since all the values we will be using are "normalized", IE: have a speed of 1
         fireballVector = fireballVector.normalize(); // you normalize it for comparison with the new direction to see if we are trying to steer too far
 
-        Block targetBlock = DirectorUtils.getDirectorBlock(p, AADirectorRange);
-        Vector targetVector;
+        // the player is looking at nothing, shoot in that general direction
+        Vector targetVector = p.getLocation().getDirection();
 
-        if (targetBlock == null || targetBlock.getType().isAir()) // the player is looking at nothing, shoot in that general direction
-            targetVector = p.getLocation().getDirection();
-        else { // shoot directly at the block the player is looking at (IE: with convergence)
-            targetVector = targetBlock.getLocation().toVector().subtract(fireball.getLocation().toVector());
-            targetVector = targetVector.normalize();
+        if (AADirectorRange >= 0) {
```

Comment:
What is the point of this check?  If you mean to check the configuration, that should be done once on load not on every fireball.

Reply:
It is a less than comparison. Sure, i could cache it in a boolean value, but that isn't really a difference. Point is to be able to actually disable converging completely. With the "original" version of yours that is NOT possible (tried setting it to negative values and 0). The only "proper" way to disable converging on your version is defining a long list of blocks that are "transparent", not really clean to manage and unnecessary. Plus, if the distance is less than or equals to 0, performing the raycast is completely unnecessary.